src/server/scripts/EasternKingdoms/zone_tirisfal_glades.cpp

```diff
@@ -38,134 +36,161 @@ EndContentData */
 
 enum Calvin
 {
-    SAY_COMPLETE        = 0,
-    SPELL_DRINK         = 2639,                             // possibly not correct spell (but iconId is correct)
-    QUEST_590           = 590
+    SAY_COMPLETE = 0,
+    SPELL_FOOD = 7737,
+    QUEST_590 = 590
+};
+
+enum event
+{
+    EVENT_EMOTE = 1,
+    EVENT_TALK = 2,
+    EVENT_DRINK_AND_QUEST_COMQUEST = 3,
+    EVENT_END = 4,
 };
 
 class npc_calvin_montague : public CreatureScript
 {
 public:
-    npc_calvin_montague() : CreatureScript("npc_calvin_montague") { }
+    npc_calvin_montague() : CreatureScript("npc_calvin_montague") {}
 
     bool OnQuestAccept(Player* player, Creature* creature, Quest const* quest) override
     {
         if (quest->GetQuestId() == QUEST_590)
         {
-            creature->SetFaction(FACTION_ENEMY);
-            creature->SetImmuneToPC(false);
-            CAST_AI(npc_calvin_montague::npc_calvin_montagueAI, creature->AI())->AttackStart(player);
+            creature->SetFaction(FACTION_MONSTER);
+            creature->RemoveNpcFlag(UNIT_NPC_FLAG_QUESTGIVER);
+            creature->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);//This is reset when out of combat
+            creature->SetUnitFlag(UNIT_FLAG_PET_IN_COMBAT);
```

Comment:
Is this sniffed? this looks weird to say the least

Reply:
```
ServerToClient: SMSG_UPDATE_OBJECT (0x27CB) Length: 169 ConnIdx: 1 Time: 12/03/2023 09:15:18.199 Number: 74959
NumObjUpdates: 2
MapID: 0 (0)
HasRemovedObjects: False
Data size: 158
[0] UpdateType: Values
[0] Object Guid: Full: 0x000000000000000000000001 Player/0 R4904/S0 Map: 0 Low: 13355676
[0] Health: 13872
[0] Flags: 526344
[0] PvpFlags: 1
[0] PlayerFlags: 256
[1] UpdateType: Values
[1] Object Guid: Full: 0x000000000000000000000001 Creature/0 R4908/S154 Map: 0 Entry: 6784 Low: 6970167
[1] EntryID: 6784
[1] DynamicFlags: 0
[1] Health: 102
[1] MaxHealth: 102
[1] DisplayID: 5509
[1] StateSpellVisualID: 0
[1] StateAnimID: 1772
[1] StateAnimKitID: 0
[1] Target: Full: 0x000000000000000000000001 Player/0 R4904/S0 Map: 0 Low: 13355676
[1] FactionTemplate: 14
[1] Flags: 559616
[1] [0] NpcFlags: 0
[1] [1] NpcFlags: 0
```
[0] Flags: 526344 and [1] Flags: 559616  Is it?

Reply:
2048 is UNIT_FLAG_PET_IN_COMBAT

Reply:
![image](https://github.com/azerothcore/azerothcore-wotlk/assets/49193927/430c4603-b199-4b7d-8029-88c81dc52a0a)